src/main/webapp/js/lfbxml/XmlBuilder.js

```diff
@@ -206,11 +206,27 @@
     const root = graph.model.getRoot();
 
     const controlPlane = findCell(root, {style: {shape: 'controlPlane'}});
-    console.log('controlPlane', controlPlane);
     if (controlPlane === null) {
       throw 'Graph has no Control Plane object.';
     }
 
+    controlPlaneStyle = parseStyle(controlPlane.style);
+    schema.name = controlPlaneStyle.name;
+    schema.classId = controlPlaneStyle.classId;
+    schema.synopsis = controlPlaneStyle.synopsis;
+    schema.version = controlPlaneStyle.version;
+
+    const verifySchemaProperties = (name, value, type) => {
+      if (name == undefined || name === null || name.trim() === '') {
+        throw `Control Plane has no name`
+      }
+      if (value == undefined || value === null || value.trim() === '') {
+        throw `Control Plane ${name} needs ${type}.`
+      }
+    }
+    verifySchemaProperties(schema.name, schema.classId, "id");
+    verifySchemaProperties(schema.name, schema.version, "version");
```

Comment:
In this specific case, you may want to treat `value` as a "truthy" value, as you're mostly doing all comparisons it already does:

```suggestion
    const ensureNotEmpty = (value, failMsg) => {
      if (!value || !value.trim()) {
        throw failMsg;
      }
    }
    ensureNotEmpty(schema.name, "Control Plane must have a name");
    ensureNotEmpty(schema.classId, `Control Plane ${schema.name} must have a Class ID`);
    ensureNotEmpty(schema.version, `Control Plane ${schema.name} must have a Version`);
```

Note that I also added a comparison for name with just `verifySchemaProperties`.

Reply:
Also you missed the `;` after `throw`.